apps/sms/views/shared/js/system_message_handler.js

```diff
@@ -329,6 +332,12 @@
       // Conversation view to the user, we remove that notification from the
       // tray that causes the second system message with "clicked" set to false.
       if (!notification.clicked || !notification.data) {
+        // Force closing the window for notification removal case
+        // because navigation didn't init correctly if pending notification
+        // message existed.
+        if (Navigation.hasPendingInit()) {
+          return Promise.reject(window.close());
```

Comment:
@julienw I would prefer and close the app more if we can make sure it's safe to do so. For this change we will need to change the hasPendingInit flag in toPanel as well.